gateway/handle.go

```diff
@@ -763,13 +764,17 @@ func (gw *Handle) extractJobsFromInternalBatchPayload(reqType string, body []byt
 	if err != nil {
 		stat.RequestFailed(response.InvalidJSON)
 		stat.Report(gw.stats)
+		gw.logger.Errorn("invalid json in request",
+			logger.NewStringField("body", string(body)),
+			logger.NewErrorField(err))
 		return nil, errors.New(response.InvalidJSON)
 	}
 	gw.requestSizeStat.Observe(float64(len(body)))
 
 	if len(messages) == 0 {
 		stat.RequestFailed(response.NotRudderEvent)
 		stat.Report(gw.stats)
+		gw.logger.Errorn("no messages in request", logger.NewStringField("body", string(body)))
```

Comment:
Again, let's not dump the body. Payloads should be encrypted and uploaded somewhere safe, not logged like this.

Reply:
Same applies to all other places where the message payload is being dumped.

Reply:
Uploading should be encrypted imo.